### PR TITLE
APIKIT-2620: Initialise AMF with custom scheduler

### PR DESCRIPTION
**Pull request by juanchib**

### CODE REVIEW

File: interface-impl-amf/src/main/java/org/mule/amf/impl/loader/ExchangeDependencyResourceLoader.java [hunk 1]
```diff
@@ -12,26 +12,32 @@
...
   }
 
   @Override
   public CompletableFuture<Content> fetch(String path) {
-    if (path == null || path.isEmpty())
-      return fail();
+    final CompletableFuture<Content> future = new CompletableFuture<>();
+
+    if (path == null || path.isEmpty()) {
+      future.completeExceptionally(new Exception("Failed to apply."));
```
Comment (JuanAller) [inline interface-impl-amf/src/main/java/org/mule/amf/impl/loader/ExchangeDependencyResourceLoader.java:38]: why not throw Exception ?

File: interface-impl-amf/pom.xml [hunk 2]
```diff
@@ -18,8 +18,8 @@
         <formatterConfigPath>../formatter.xml</formatterConfigPath>
         <sonar.jacoco.reportPaths>../target/jacoco.exec</sonar.jacoco.reportPaths>
 
-        <amf.version>4.0.6</amf.version>
-        <amf.xml.extension.version>1.4.1</amf.xml.extension.version>
+        <amf.version>4.1.0-2-INTERNAL-SNAPSHOT</amf.version>
```
Comment (JuanAller) [inline interface-impl-amf/pom.xml:21]: Can we use syncValidate in XML payloads ? using this AMF version

Comment (juanchib) [reply interface-impl-amf/pom.xml:21]: extension should be 1.6.0

Comment (JuanAller) [reply interface-impl-amf/pom.xml:21]: sure `<amf.xml.extension.version>1.6.0</amf.xml.extension.version>`

Comment (JuanAller) [reply interface-impl-amf/pom.xml:21]: my comment was in the previous line , sorry 😆

Comment (juanchib) [reply interface-impl-amf/pom.xml:21]: will do !

File: interface-impl-amf/src/main/java/org/mule/amf/impl/loader/ExchangeDependencyResourceLoader.java (hunk 1 above)
Comment (juanchib) [reply interface-impl-amf/src/main/java/org/mule/amf/impl/loader/ExchangeDependencyResourceLoader.java:38]: https://www.mulesoft.org/jira/browse/APIKIT-2599

File: parser-service/src/main/java/org/mule/parser/service/strategy/WithFallbackParsingStrategy.java [hunk 3]
```diff
@@ -19,17 +20,29 @@
...
   @Override
   public ParseResult parse(ApiReference ref) {
     ParseResult amfResult = AMF_DELEGATE.parse(ref);
     if (amfResult.success()) {
       return amfResult;
     }
-    ParseResult ramlResult = new RamlParsingStrategy(new ReferencesResolver(amfResult)).parse(ref);
+    ReferencesResolver referencesResolver = new ReferencesResolver(amfResult);
+
+    if(executor == null){
```
Comment (dcourtil) [inline parser-service/src/main/java/org/mule/parser/service/strategy/WithFallbackParsingStrategy.java:33]: Shouldn't be ```!=```?

Comment (juanchib) [reply parser-service/src/main/java/org/mule/parser/service/strategy/WithFallbackParsingStrategy.java:33]: yep

**Pull request merged**